### ISSUE TITLE
Add top-level assertWith() as alternative to satisfies() for local variables

### ISSUE THREAD

**Issue by jodastephen** [assertj/assertj-core#1535]
#### Summary

The ability to group together tests using `satisfies()` is very useful and better than the `returns()` fluent expression in most cases. It can shorten some use cases quite significantly. However, I didn't find it immediately, and I think the name is not great when not sharing a condition. Could I suggest an alternative top-level entry point, `assertWith(Object, Consumer)` that would make this more obvious?

#### Example

Current:

```java
assertThat(team.getPlayers().get(0).getStats()).satisfies(t-> {
   assertThat(t.pointPerGame).isGreaterThan(25.7);
   assertThat(t.assistsPerGame).isGreaterThan(7.2);
   assertThat(t.reboundsPerGame).isBetween(9, 12);
});
```

Proposed:

```java
assertWith(team.getPlayers().get(0).getStats(), t-> {
   assertThat(t.pointPerGame).isGreaterThan(25.7);
   assertThat(t.assistsPerGame).isGreaterThan(7.2);
   assertThat(t.reboundsPerGame).isBetween(9, 12);
});
```

(I'd also be happy with `with`, `using` or `assertUsing`.

**Comment by joel-costigliola** [comment]
I'm ok to add `assertWith` if it makes it easier to group assertions. One disadvantage of this top level method is that they can't be used with [soft assertions](http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#soft-assertions).

I find `satisfies` to be a good name (well we could not find a better one), the only problem being that it is the same for `Condition`. Is that why you did not like it ? I'm not sure I have understood your point.

**Comment by jodastephen** [comment]
I don't think soft assertions are relevant here, as the `assertWith` doesn't actually do anything. Users that wanted to soft assert could still do so within the `assertWith` block.

`satisfies` is a good name when the lambda it is passed is a variable and reused multiple times:

    assertThat(team.getPlayers().get(0).getStats()).satisfies(pointsInStandardRange);
    assertThat(team.getPlayers().get(1).getStats()).satisfies(pointsInStandardRange);

But it is a less good name when you simply want to group some things together inline. In some ways `assertWith` is an imperfect way to get a language change. Groovy has [with](https://mrhaki.blogspot.com/2009/09/groovy-goodness-with-method.html) blocks which turns the target into the equivalent of `this`. And this is why the suggestion uses the name `assertWith`, to match similar features in other languages. (Although `with()` might be clearer to read, it is so short I worried it might clash with other projects, hence I suggested `assertWith`).
